libtiledbvcf/src/read/reader.cc

```diff
@@ -1427,6 +1426,16 @@ void Reader::info_attribute_count(int32_t* count) {
   *count = this->dataset_->info_field_types().size();
 }
 
+void Reader::sample_count(int32_t* count) {
+  if (count == nullptr)
+    throw std::runtime_error("Error getting sample count");
+  *count = dataset_->metadata().sample_names.size();
+}
+
+void Reader::sample_name(int32_t index, const char** name) {
+  *name = dataset_->metadata().sample_names[index].c_str();
```

Comment:
This will result in `name` having an invalid string reference. The `c_str()` is short lived and not guaranteed beyond the line it is called on.